src/foam/nanos/pii/PIIReportTicketSendRuleAction.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license
+ * Copyright 2025 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.pii',
+  name: 'PIIReportTicketSendRuleAction',
+
+  documentation: 'Generate email from report and send.',
+
+  implements: [
+    'foam.nanos.ruler.RuleAction'
+  ],
+
+  javaImports: [
+    'foam.core.ContextAgent',
+    'foam.core.FObject',
+    'foam.core.X',
+    'foam.dao.DAO',
+    'static foam.mlang.MLang.*',
+    'foam.nanos.auth.User',
+    'foam.nanos.crunch.UserCapabilityJunction',
+    'foam.nanos.fs.File',
+    'foam.nanos.logger.Logger',
+    'foam.nanos.logger.Loggers',
+    'foam.nanos.logger.PrefixLogger',
+    'foam.nanos.notification.email.EmailMessage',
+    'foam.nanos.notification.email.EmailPropertyService',
+    'foam.nanos.session.Session',
+    'foam.nanos.ticket.Ticket',
+    'foam.util.SafetyUtil',
+    'java.util.Arrays',
+    'java.util.HashMap',
+    'java.util.List',
+    'java.util.Map'
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'emailTemplate',
+      value: 'foam-nanos-pii-report-EmailTemplate',
+    }
+  ],
+
+  methods: [
+    {
+      name: 'applyAction',
+      javaCode: `
+        final PIIReportTicket ticket = (PIIReportTicket) obj;
+        if ( ticket.getDocuments() == null ||
+             ticket.getDocuments().length == 0 ) {
+          // TODO: how to report to user.
```

Comment:
Should probably just send the email anyways but make the template have some message about there being no documents?

Reply:
or if this should truly never happen maybe raise an alarm?

Reply:
Yeah, pii should only be enabled for systems with full user data - names, email, other onboarding documents so there will always be at least the keyvalue pdf.  
Any errors/exceptions are system configuration related - and it now generates an eventrecord which uses notification to alert operations.